### ISSUE TITLE
Top icon link to localhost:5000 when actually its being hosted in a different hostname and behind nginx

### ISSUE THREAD

**Issue by goodwill** [getredash/redash#2498]
### Issue Summary

The top red chart icon links to 127.0.0.1:5000 when actually its hosted behind proxy, suggest to add ENV setup to allow hostname to be put in.

### Technical details:

* Redash Version: Redash 4.0.0+b3948
* Browser/OS: All
* How did you install Redash: docker-compose + nginx on the host machine

**Comment by kravets-levko** [comment]
Hi @goodwill! Thank you for reporting this issue. We already know about it, but it's a bit more complicated than just wrong logo link - we have the same issue with embed/share URLs. Currently we're trying to find a solution that will fix URLs for all cases.

**Comment by arikfr** [comment]
@goodwill,

The URL for the icon is generated by the server. The server deduces its address using the `Host` header. This works fine with proxies as long as your proxy is configured to pass the correct header.

With nginx you need to make sure that you have the following in your config:

```
proxy_set_header Host $http_host;
```

**Comment by arikfr** [comment]
@kravets-levko is the issue you're referring is what we had when running with the `webpack-dev-server`?

**Comment by kravets-levko** [comment]
@arikfr Yes, exactly, but it's the same case - `webpack-dev-server` is a proxy to a backend.

**Comment by arikfr** [comment]
@kravets-levko it happens for the same reason: not setting the `Host` header correctly. For the webpack-dev-server proxy, we set [`changeOrigin` to `true`](https://github.com/getredash/redash/blob/master/webpack.config.js#L190), which means that it will send the destination's hostname as the value of the `Host` header.

Setting this setting to `false` will work with a regular usage of the wds proxy (when proxying locally to localhost), and you will see that the URLs are correct (prefixed with `localhost:8080`). But in most cases it won't work with a remote server, unless it ignores the Host header.

I think we should change this value to `false` and update the configuration of our dev server to have a default server setup, so it will work with this configuration.
